app/src/main/java/org/mozilla/fenix/settings/AccountSettingsFragment.kt

```diff
@@ -97,19 +110,81 @@ class AccountSettingsFragment : PreferenceFragmentCompat(), CoroutineScope {
         }
 
         // Sync stopped successfully.
-        override fun onIdle() {
+        override fun onIdle(result: SyncResult) {
             CoroutineScope(Dispatchers.Main).launch {
                 val pref = findPreference<Preference>(context!!.getPreferenceKey(R.string.pref_key_sync_now))
                 pref?.let {
                     pref.title = getString(R.string.preferences_sync_now)
                     pref.isEnabled = true
                     updateLastSyncedTimePref(context!!, pref, failed = false)
                 }
+
+                Log.i(LOGTAG, "Sync finished; recording telemetry")
+                for ((_, status) in result) {
+                    status.stats?.let {
+                        it.syncs.forEach {
+                            it.engines.forEach {
+                                when (it.name) {
+                                    "history" -> {
+                                        Log.i(LOGTAG, "Got history sync")
+                                        HistorySync.apply {
+                                            startedAt.set(Date(it.at.toLong()))
+                                            // Glean doesn't support recording arbitrary durations in
+                                            // timespans, so we record absolute start and end times instead.
+                                            finishedAt.set(Date((it.at + it.took).toLong()))
+                                            incoming["applied"].set(it.incoming?.applied ?: 0)
```

Comment:
Do you really care about storing `0`? Glean will not send fields for which it has no value stored (the same consideration applies for the other fields below).